progenitor-client/src/progenitor_client.rs

```diff
@@ -118,6 +118,14 @@ impl<T> ResponseValue<T> {
         &self.headers
     }
 
+    /// Get the parsed value of the Content-Length header, if present and valid.
+    pub fn content_length(&self) -> Option<u64> {
+        self.headers
+            .get(reqwest::header::CONTENT_LENGTH)
+            .and_then(|cl| cl.to_str().ok())
+            .and_then(|cl| cl.parse::<u64>().ok())
```

Comment:
Is this the same as this?

```rust
        self.headers
            .get(reqwest::header::CONTENT_LENGTH)?
            .to_str()
            .ok()?
            .parse::<u64>()
            .ok()
 ```

I'm fine either way... just offering that up in case you find it to be more legible.

Reply:
Ah, yeah, that's probably better.  I forget that they did `?` for `Option` provided you aren't doing any `Result` business as well.